### PR TITLE
Fix UK Hash and Backslash mixup

### LINKED ISSUE #25: UK ScanCodeSet 2 errors
On a PS/2 UK keyboard with 0.6.1:

* The `#~` key produces `\|`
* The `\|` key produces `#~`

### PR DESCRIPTION
**Pull request by thejpster**
Closes #25

### CODE REVIEW

File: src/layouts/uk105.rs [hunk 1]
```diff
@@ -53,14 +53,67 @@ impl KeyboardLayout for Uk105Key {
                     DecodedKey::Unicode('4')
                 }
             }
-            KeyCode::HashTilde => {
+            KeyCode::BackSlash => {
```
Comment (ryan-summers) [inline src/layouts/uk105.rs:56]: Can you add the docs that you have in the original issue? I pulled up pictures of the UK105 keyboard and was still confused about why this change was necessary. Maybe link back to the original issue as well?

**Review by ryan-summers — COMMENTED** [review]
Minor doc request, otherwise looks good to me :) Also needs  CHANGELOG entry

**Comment by thejpster** [comment]
Rather than explain why HashTilde and Backslash are swapped on a UK keyboard, I got a bit carried away and fundamentally reworked how we name Key Codes.

**Review by ryan-summers — APPROVED** [review]
I like the changes, but the `Oem` codes are a bit opaque to me. Which makes sense if every OEM uses them differently.

In any case, I like the changes here. I haven't thoroughly reviewed every key mapping, but I imagine it's easy for us to fix issues as they arise.

**Pull request merged**